pkg/k8s/client.go

```diff
@@ -66,31 +67,47 @@ func NewClient(log logrus.FieldLogger, url string, kubeConfig string) (Client, e
 	}, nil
 }
 
-// GetKubernetesClient is used to get the kubernetes clientset.
-func (w *ClientWrapper) GetKubernetesClient() kubeclient.Interface {
+// buildConfig takes the master URL and kubeconfig, and returns an external or internal config.
+func buildConfig(log logrus.FieldLogger, masterURL, kubeConfig string) (*rest.Config, error) {
+	if os.Getenv("KUBERNETES_SERVICE_HOST") != "" && os.Getenv("KUBERNETES_SERVICE_PORT") != "" {
+		// If these env vars are set, we can build an in-cluster config.
+		log.Infoln("Creating in-cluster client")
+		return rest.InClusterConfig()
+	}
+
+	if masterURL != "" && kubeConfig != "" {
```

Comment:
After thinking a bit more about it, I think that the `masterURL` is only used as a config [override](https://github.com/kubernetes/client-go/blob/master/tools/clientcmd/client_config.go#L569) and is not mandatory (e.g. I'm using only the `kubeconfig` flag when I'm running Maesh binary). So the test should look like my suggestion and error messages should be updated accordingly.

```suggestion
	if kubeConfig != "" {
```

WDYT?

Reply:
Turns out I do the opposite! I only use MasterURL, since I use kubectl proxy :D

I should make this `&&` an `||`, and update that message.